doc/releases/changelog-dev.md

```diff
@@ -35,6 +35,14 @@
   linking against Python shared libraries after decoupling Python from the Runtime C-API.
   [(#1305)](https://github.com/PennyLaneAI/catalyst/pull/1305)
 
+* Measurement primitives now support dynamic shape at the frontend,
+  although at the PennyLane side, the corresponding operations still lack such support.
+
+  To support this, the measurement operations now take in shots as a proper argument
+  instead of an integer attribute with a literal integer value.
+  [(#1170)](https://github.com/PennyLaneAI/catalyst/pull/1170)
+  [(#1310)](https://github.com/PennyLaneAI/catalyst/pull/1310)
```

Comment:
Just a suggestion; feel free to modify if I've missed your intention.

```suggestion
* Measurement primitives now support dynamic shapes in the frontend.
  [(#1170)](https://github.com/PennyLaneAI/catalyst/pull/1170)
  [(#1310)](https://github.com/PennyLaneAI/catalyst/pull/1310)

  Note that the corresponding operations in PennyLane currently do not support dynamic shapes. To
  enable this functionality, measurement operations now accept shots as an argument instead of an
  integer attribute with a literal integer value.
```

Reply:
This is Raul's description from the original PR. I will rewrite it with the new changes later.

Reply:
The changelog now is more or less completely rewritten.